aiida/tools/visualization/graph.py

```diff
@@ -0,0 +1,727 @@
+    process_map = {
+        "process.calculation.calcjob.CalcJobNode.": {
+            "style": "filled",
+            "shape": "rectangle",
+            "pencolor": "black",
+            "fillcolor": "#87CEFA"  # light blue
+        },
+        "process.calculation.calcfunction.CalcFunctionNode.": {
+            "style": "filled",
+            "shape": "parallelogram",
+            "pencolor": "black",
+            "fillcolor": "#87CEFA"  # light blue
+        },
+        "process.workflow.workchain.WorkChainNode.": {
+            "style": "filled",
+            "shape": "octagon",
+            "pencolor": "black",
+            "fillcolor": "#FFA500"
+        },
+        "process.workflow.workfunction.WorkFunctionNode.": {
+            "style": "filled",
+            "shape": "hexagon",
+            "pencolor": "black",
+            "fillcolor": "#FFA500"  # light blue
+        }
+    }
+
+    node_style = process_map.get(class_node_type, default)
+
+    # style process node, based on success/failure of process
+    if node.is_failed:
+        node_style['penwidth'] = "3.0"
+        node_style['color'] = '#FF0000'  # red
+    elif node.is_finished_ok:
+        node_style['penwidth'] = "3.0"
+        node_style['color'] = '#008000'  # green
+
+    return node_style
+
+
+def _add_graphviz_node(graph,
+                       node,
+                       data_style_func,
+                       process_style_func,
+                       data_sublabel_func,
+                       style_override=None,
+                       include_sublabels=True):
+    """create a node in the graph
+
+    :param graph: the graphviz.Digraph to add the node to
+    :param node: the node to add
+    :type node: aiida.orm.nodes.node.Node
+    :param data_style_func: callable mapping a data node instance to a dictionary defining the graphviz node style
+    :param process_style_func: callable mapping a process node instance to a dictionary defining the graphviz node style
+    :param data_sublabel_func: callable mapping a node instance to a sub-label for the node text
+    :param style_override: style dictionary, whose keys will override the final computed style (Default value = None)
+    :type style_override: None or dict
+    :param include_sublabels: whether to include the sublabels for nodes (Default value = True)
+    :type include_sublabels: bool
+
+    nodes are styled based on the node type
+
+    For subclasses of Data, the ``class_node_type`` attribute is used
+    for mapping to type specific styles
+
+    For subclasses of ProcessNode, we choose styles to distinguish between
+    types, and also color the nodes for successful/failed processes
+
+    The first line of the node text is always '<node.name> (<node.pk>)'.
+    Then, if ``include_sublabels=True``, subsequent lines are added,
+    which are node type dependant.
+
+    """
+    # pylint: disable=too-many-arguments
+    node_style = {}
+    if isinstance(node, BaseFactory("aiida.node", "data")):
```

Comment:
Since this is using entry points, I'm not sure whether it's a good idea to create the factory for every node... will this be cached?
Anyhow, you could also just `from aiida.orm import Data`